cmd/cored/main.go

```diff
@@ -191,12 +187,7 @@ func main() {
 		chainlog.Fatalkv(ctx, chainlog.KeyError, errors.Wrap(err, "Serve"))
 	}()
 
-	if useTLS {
-		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{
-			Certificates: []tls.Certificate{*cert},
-			RootCAs:      loadRootCAs(*rootCAs),
-		}
-	}
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsConfig
```

Comment:
wdyt about ferrying the config into the `chain/core` package, creating a new `http.Client` and installing it at  `core.API.httpClient`? Then we'd consistently use  `core.API.httpClient` in tests and in production.

I think manipulating `http.DefaultTransport` here could be a race with `raft.Start` which might already be using `http.DefaultClient` to make RPCs to other nodes?